src/controllers/dynakube/activegate/internal/tenantinfo/reconciler_test.go

```
@@ -0,0 +1,97 @@
+)
+
+var (
+	tenantInfoResponse = &dtclient.ActiveGateTenantInfo{
+		TenantInfo: dtclient.TenantInfo{
+			UUID:  "testUUID",
+			Token: "dt.some.valuegoeshere",
+		},
+		Endpoints: "someEndpoints",
+	}
+)
+
+func newTestReconciler(client client.Client) *Reconciler {
+	instance := &dynatracev1beta1.DynaKube{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: testNamespace,
+			Name:      testDynakubeName,
+		},
+		Spec: dynatracev1beta1.DynaKubeSpec{
+			APIURL: "https://testing.dev.dynatracelabs.com/api",
+		},
+	}
+	dtc := &dtclient.MockDynatraceClient{}
+	dtc.On("GetActiveGateTenantInfo", mock.Anything).Return(tenantInfoResponse, nil)
+
+	r := NewReconciler(client, client, scheme.Scheme, instance, dtc)
+	return r
+}
+
+func TestReconcile(t *testing.T) {
+	t.Run(`reconcile tenant info for first time`, func(t *testing.T) {
+		r := newTestReconciler(fake.NewClientBuilder().Build())
+		update, err := r.Reconcile()
+
+		var tenantInfoSecret corev1.Secret
+		r.Client.Get(context.TODO(), client.ObjectKey{Name: extendWithAGSecretSuffix(r.dynakube.Name), Namespace: testNamespace}, &tenantInfoSecret)
+
+		assert.Equal(t, []byte(tenantInfoResponse.UUID), tenantInfoSecret.Data[TenantUuidName])
+		assert.Equal(t, []byte(tenantInfoResponse.Token), tenantInfoSecret.Data[TenantTokenName])
+		assert.Equal(t, []byte(tenantInfoResponse.Endpoints), tenantInfoSecret.Data[CommunicationEndpointsName])
+		assert.True(t, update)
+		assert.NoError(t, err)
+	})
+
+	t.Run(`reconcile tenant info changed`, func(t *testing.T) {
+		r := newTestReconciler(fake.NewClientBuilder().Build())
+		update, err := r.Reconcile()
+		assert.True(t, update)
+		assert.NoError(t, err)
+
+		var newTenantToken = "dt.someOtherToken"
+		tenantInfoResponse.UUID = newTenantToken
+
+		update, err = r.Reconcile()
+
+		var tenantInfoSecret corev1.Secret
+		r.Client.Get(context.TODO(), client.ObjectKey{Name: extendWithAGSecretSuffix(r.dynakube.Name), Namespace: testNamespace}, &tenantInfoSecret)
```

same as bevore, do not implicitly ignore an error